dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowVersionDAO.java

```diff
@@ -124,15 +129,55 @@ private List<Predicate> processQuery(String sortCol, String sortOrder, CriteriaB
                 sortExpression = version.get(sortCol);
             }
         } else {
+            // Create a sort expression that "scores" each version.
+            // To ensure that "relevant" versions appear first, we rank "mainstem" branches, tags, more recently-modified versions, valid versions, and versions with metrics higher.
+            Path<?> name = version.get("name");
+            Path<?> referenceType = version.get("referenceType");
+            Path<Boolean> valid = version.get("valid");
+            Path<? extends Collection> metricsByPlatform = version.get("metricsByPlatform");
+
+            // Compute the "age" of the version, which is the date since it was last modified, in days.
+            // To prevent the sort order from constantly changing, calculate the age relative to the beginning of tomorrow.
+            Expression<Double> secondsPerDay = cb.literal(24 * 3600.);
+            Expression<Timestamp> now = cb.currentTimestamp();
+            Expression<Timestamp> modified = cb.coalesce(lastModified, version.get("dbCreateDate"));
+            Expression<Double> nowDays = cb.toDouble(cb.quot(cb.function("date_part", Double.class, cb.literal("epoch"), now), secondsPerDay));
+            Expression<Double> modifiedDays = cb.toDouble(cb.quot(cb.function("date_part", Double.class, cb.literal("epoch"), modified), secondsPerDay));
+            Expression<Double> tomorrowDays = cb.sum(cb.function("floor", Double.class, nowDays), 1.);
+            Expression<Double> ageDays = cb.diff(tomorrowDays, modifiedDays);
+
+            // Calculate some weights based on the "age", validity, and presence of metrics.
+            Expression<Double> ageWeight = cb.toDouble(cb.quot(1., cb.function("greatest", Double.class, ageDays, cb.literal(0.5))));
+            Expression<Double> validWeight = cb.<Double>selectCase()
+                .when(cb.isTrue(valid), 3.)
+                .otherwise(1.);
+            Expression<Double> metricsWeight = cb.<Double>selectCase()
+                .when(cb.isNotEmpty(metricsByPlatform), 2.)
+                .otherwise(1.);
+
+            // Calculate a base "score", which ranks "mainstem" branches highest and tags higher.
+            Expression<Double> baseScore = cb.<Double>selectCase()
+                .when(cb.equal(name, "master"), 20.)
```

Comment:
could be ticket, we could also grab what they set as the default branch for their repo, it may be in our database already